src/aggregated_range_proof/mod.rs

```diff
@@ -1,122 +1,187 @@
 #![allow(non_snake_case)]
 
-pub mod dealer;
-pub mod messages;
-pub mod party;
+use rand::Rng;
 
-#[cfg(test)]
-mod tests {
-    use std::iter;
+use curve25519_dalek::scalar::Scalar;
 
-    use rand::{OsRng, Rng};
+use generators::Generators;
+use proof_transcript::ProofTranscript;
 
-    use curve25519_dalek::scalar::Scalar;
-    use proof_transcript::ProofTranscript;
-
-    use super::dealer::*;
-    use super::messages::*;
-    use super::party::*;
+pub mod dealer;
+pub mod messages;
+pub mod party;
 
-    fn create_multi<R: Rng>(
+pub use self::messages::AggregatedProof;
+
+struct SinglePartyAggregator {}
+
+impl SinglePartyAggregator {
+    /// Create an aggregated rangeproof of multiple values.
+    ///
+    /// This performs the same proof aggregation MPC protocol, but
+    /// with one party playing all roles.
+    ///
+    /// The length of `values` must be a power of 2.
+    ///
+    /// XXX this should allow proving about existing commitments.
+    fn generate_proof<R: Rng>(
+        generators: &Generators,
+        transcript: &mut ProofTranscript,
         rng: &mut R,
-        values: Vec<u64>,
+        values: &[u64],
         n: usize,
-    ) -> (Proof, Vec<ProofShareVerifier>) {
-        use generators::{Generators, PedersenGenerators};
+    ) -> Result<AggregatedProof, &'static str> {
```

Comment:
Can it actually ever fail? Maybe we can do `unwrap()` inside the implementation to avoid exposing unused errors?